src/DotLiquid/Tokenizer.cs

```diff
@@ -90,13 +99,68 @@ internal static List<string> Tokenize(string source)
             return tokens;
         }
 
+        /// <summary>
+        /// Enumerates over a variable sequence in dotted or bracket notation
+        /// </summary>
+        /// <param name="source">The Liquid Variable string</param>
+        /// <returns></returns>
+        /// <exception cref="SyntaxException"></exception>
+        internal static IEnumerator<string> GetVariableEnumerator(string source)
+        {
+            if (string.IsNullOrEmpty(source))
+                yield break;
+
+            using (var markupEnumerator = new CharEnumerator(source))
```

Comment:
This using lasts until the end of method so it can be simplified as

`using var markupEnumerator = new CharEnumerator(source);`

and no code block.

Reply:
This won't compile for .net 4.5 or core 1.x - 
Feature 'using declarations' is not available in C# 7.3. Please use language version 8.0 or greater.